eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php

```diff
@@ -1052,6 +1052,55 @@ public function testAssignSection()
     }
 
     /**
+     * Test search engine is updated after removal of the translation from all the Versions.
+     */
+    public function testRemoveTranslation()
+    {
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+        $contentService = $repository->getContentService();
+
+        $content = $this->createMultiLanguageContent(
+            [
+                'eng-US' => 'AmE Name',
+                'eng-GB' => 'BrE Name',
+            ],
+            2,
+            false
+        );
+
+        $contentService->removeTranslation($content->contentInfo, 'eng-GB');
+
+        $this->refreshSearch($repository);
+
+        // Test ContentId search returns Content without removed Translation
+        $query = new Query([
+            'query' => new Criterion\ContentId($content->contentInfo->id),
+            'filter' => new Criterion\LanguageCode('eng-GB', false),
+        ]);
+        $result = $searchService->findContent($query);
+        self::assertEquals(0, $result->totalCount);
+
+        // Test FullText search for removed unique name part returns no results
+        $query = new Query([
+            'query' => new Criterion\FullText('BrE'),
+        ]);
+        $result = $searchService->findContent($query);
+        self::assertEquals(0, $result->totalCount);
+
+        if (!$this->getSetupFactory() instanceof LegacyElasticsearchSetupFactory) {
+            // Test Location Search returns Content without removed Translation
+            $query = new LocationQuery(
+                [
+                    'query' => new Criterion\FullText('BrE'),
```

Comment:
> If it's the latter case, then maybe ES dispatcher should not throw exception? 

won't be the same as with facets, User won't have any hint of why his criterion is not taken into account in this case. So -1, we stick with this as is, and cleanup once ES matures.